django/db/models/sql/compiler.py

```diff
@@ -802,6 +809,35 @@ def get_related_klass_infos(klass_info, related_klass_infos):
                 )
         return related_klass_infos
 
+    def get_select_for_update_of_arguments(self):
+        """
+        Return a quoted list of arguments to the SELECT FOR UPDATE OF
+        part of the query.
+        """
+        result = []
+
+        for name in self.query.select_for_update_of:
+            parts = [] if name == 'self' else name.split(LOOKUP_SEP)
+            klass_info = self.klass_info
+            for part in parts:
+                for related_klass_info in klass_info.get('related_klass_infos', []):
+                    if related_klass_info['field'].name == part:
+                        klass_info = related_klass_info
+                        break
+                else:
+                    raise FieldError(
+                        "Bad field given in select_for_update(of={...}): '%s'. "
+                        "Only relational fields followed in the query are allowed." % name
+                    )
```

Comment:
I wonder if we could enumerate valid options like we do for `select_related()`/`prefetch_related()`.